src/index.js

```diff
@@ -30,7 +28,9 @@ async function listen(credentials, notificationCallback) {
   if (!credentials.keys.authSecret) {
     throw new Error('Missing keys.authSecret in credentials');
   }
-  await checkIn(credentials.gcm.androidId, credentials.gcm.securityToken);
-  emitter.on(ON_NOTIFICATION_RECEIVED, notificationCallback);
-  await connect(credentials.gcm, credentials.keys, credentials.persistentIds);
+
+  const client = new Client(credentials, credentials.persistentIds);
+  client.on('ON_NOTIFICATION_RECEIVED', notificationCallback);
+  client.connect();
+  return client;
```

Comment:
Since client is returned here you can get #7 for free by calling client.destroy()